### DISCUSSION TITLE
Training without patches

### DISCUSSION THREAD

**Discussion by learningdeep2023** [Project-MONAI/MONAILabel#1179] [Q&A]
I am using the Radiology App with segmentation_spleen to segment one structure from CT images.
I would like to train a 3D UNET without patches/ using all available data say 256x256x32.
If possible cropped from the center of the orig images.

I would appreciate any advice on what needs to be changed in the config/infer/train/etc files.

**Comment by tangy5** [comment]
You are right, change the config/infer/train segmetnation_spleen scripts for modification. Specifically, change the 
https://github.com/Project-MONAI/MONAILabel/blob/3148ff0bc2e2043eae21a7e13e536c9482550b90/sample-apps/radiology/lib/infers/segmentation_spleen.py#L65

for inference ROI, if your model is trained with different image size. 
Change the pre-transform accordingly as well to accommodate your own training/inference.

**Comment by learningdeep2023** [comment]
Thanks for your response.

I want to confirm that "pixdim=(1.0, 1.0, 1.0)," means no interpolation.....
and pixdim=(1.0, 1.0, 0.5) means interpolation 2x in the slice dimension?

**Reply by tangy5** [reply]
If the input image is in pixdim of 1.0, 1.0, 1.0, then this transform will not do interpolation, otherwise, it will resample the input image to 1.0, 1.0, 1.0. Generally, raw CT/MRI image will not have this high resolution, this transform up-samples the images. 

Yes, 0.5 in 3rd dimension will further up-sample the image in slice thickness. (0.5mm)

**Comment by learningdeep2023** [comment]
Thanks... want to confirm the units. 1.0 is definitely mm for all pixdim's?  Not relative pixel width?

**Reply by tangy5** [reply]
Typically yes, but not necessary. The pixdim is : voxel width along dimension.
You could do some search on this parameter.

**Comment by learningdeep2023** [comment]
I think that I almost have it working.
Training is OK.
After inference, I feel that the prediction is not placed correctly on the orig image.
I used CenterSpatialCropd as a pretransform, but am unsure how to reverse this in the post transform.

Any suggestions?


def pre_transforms(self, data=None) -> Sequence[Callable]:
        return [
            LoadImaged(keys="image"),
            EnsureTyped(keys="image", device=data.get("device") if data else None),
            EnsureChannelFirstd(keys="image"),
           # Spacingd(keys="image", pixdim=[1.0, 1.0, 1.0]),
            ScaleIntensityRanged(keys="image", a_min=-57, a_max=130, b_min=0.0, b_max=1.0, clip=True),
	    CenterSpatialCropd(keys="image",roi_size=(320,320,32))   ,
          #  SelectItemsd(keys=("image")),
	             
        ]

    def inferer(self, data=None) -> Inferer:
        return SimpleInferer()

    def post_transforms(self, data=None) -> Sequence[Callable]:
        return [
            EnsureTyped(keys="pred", device=data.get("device") if data else None),
            Activationsd(keys="pred", softmax=True),
            AsDiscreted(keys="pred", argmax=True),
            Restored(keys="pred", ref_image="image"),
        ]

**Reply by tangy5** [reply]
hI @learningdeep2023 , You can see there is a "Restored" in the post transform, it will invert resize the image based on the ref_image , also MONAI Label basic infer will automatically do invert transform based on pre-transforms. Looking at above code, I see Spacingd is removed and CenterSpatialCropd is added. 

You can debug here: 
 
https://github.com/Project-MONAI/MONAILabel/blob/caa1d8d4beda805e7118b53fa8f0dcad8729faa3/monailabel/tasks/infer/basic_infer.py#L306

and here:

https://github.com/Project-MONAI/MONAILabel/blob/caa1d8d4beda805e7118b53fa8f0dcad8729faa3/monailabel/tasks/infer/basic_infer.py#L383

to double check whether the added crop is correctly inverted(pad).

Thanks.
